Deprecate the External Configuration Volumes (#10566)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/proc-loading-config-from-file.adoc

    config.providers: file # <1>
    config.providers.file.class: org.apache.kafka.common.config.provider.FileConfigProvider # <2>
  #...
  template:
    pod:
      volumes:
        - name: connector-config-volume # <3>
          secret:
            secretName: mysecret # <4>
    kafkaContainer:
      volumeMounts:
        - name: connector-config-volume # <5>
          mountPath: /mnt/mysecret # <6>
----
<1> The alias for the configuration provider is used to define other configuration parameters.
<2> `FileConfigProvider` provides values from properties files.
The parameter uses the alias from `config.providers`, taking the form `config.providers.${alias}.class`.
<3> The name of the volume containing the secret.
<4> The name of the secret.
<5> The name of the volume that will be mounted.

> Has to be the same as the name of the volume.

Reviewer: I didn't get it, but it could be an English problem. The first sentence says "The name of the volume that will be mounted". Then as I interpret the second one is ... "(It) Has to the be the same as the name of the volume" where "It" sounds to be the subject of the previous sentence which is "the name of the volume", so it looks to me like ... "the name of the volume has to be the same as the name of the volume".

Author: @ppatierno I tried to improve it, please have a look. What I'm trying to make clear here is that the name cannot be chosen randomly but that it is a name of the volume from the volume list.

Reviewer: It's clearer now. Thanks!

> <6> The path where the Secret will be mounted.
> Has to start with `/mnt/`.

Reviewer 2: ```suggestion
<6> The path where the secret is mounted, which must start with `/mnt/`.
```

  config:
    database.hostname: 192.168.99.1
    database.port: "3306"

>     database.user: "${file:/mnt//mysecret/connector.properties:dbUsername}"

Reviewer 3: ```suggestion
    database.user: "${file:/mnt/mysecret/connector.properties:dbUsername}"
```
